usefulUtilities/botPlayer/assignmentClientManager.js

```diff
@@ -0,0 +1,248 @@
+"use strict";
+
+//
+//  Bot Player
+//  assignmentClientManager.js
+//  Created by Milad Nazeri on 2019-06-06
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+(function() {
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+    // #region UTILITY FUNCTIONS   
+    
+
+    // Start playing sequence to fill players with bots
+    var AC_AVAILABILITY_CHECK_MS = 1000;
+    function startSequence() {
+        // Check to see how many bots are needed
+        if (botCount >= botsFound + 1) {
+            return;
+        }
+
+        if (botCount < availableAssignmentClientPlayers.length) {
+            var player = availableAssignmentClientPlayers[botCount];
+            player.play();
+            botCount++;
+
+            if (botCount >= botsFound + 1) {
+                return;
+            }
+        }
+
+        Script.setTimeout(function() {
```

Comment:
why is this necessary?

Reply:
This is checking to see if all there are enough assignment clients for how many bots are needed.  Since there is no way to find out how many of this script are available, we need to keep polling to see if the number of bots needed === the number of available assignment clients